### PR TITLE
Make Docker Optional

### PR DESCRIPTION
**Pull request by Aathish04**
[As discussed on Discord](https://discord.com/channels/581738731934056449/695188550836289546/894526054281129994), it would be convenient for those users to run the bot without using Docker.

This PR ensures that the bot doesn't use Docker if the `NO_DOCKER` ~environment~ variable is set to `True` in `config.py`.
~~If `NO_DOCKER` is True, it also checks i~~ If the `USE_ONLINETEX` ~~environment~ variable is `True` ~~in which case~~ it will make use of the library `manim-onlinetex` to render LaTeX as opposed to using system LaTeX, only if NO_DOCKER is also true.

NOTE:
If `USE_ONLINETEX` is true, the user will have to create a `manim.cfg` setting `TEX_DIR` to a non-temporary directory, so the bot won't have to strain external LaTeX rendering APIs.

### CODE REVIEW

File: cogs/manimate.py [hunk 1]
```diff
@@ -7,11 +7,24 @@
 from pathlib import Path
 
 import discord
-import docker
 from discord.ext import commands
 
-dockerclient = docker.from_env()
+# If you don't want to use docker, set NO_DOCKER to True
+# where-ever you define environment variables.
+if "NO_DOCKER" in os.environ:
```
[the remark this answers is not in the document]
Comment (Aathish04) [reply cogs/manimate.py]: Done :)

File: cogs/manimate.py [hunk 2]
```diff
@@ -83,15 +96,42 @@ def construct_reply(arg):
...
+                            reply_args = {
+                                "content": "Something went wrong, here is "
+                                "what Manim reports:\n",
+                                "file": discord.File(
+                                    fp=io.BytesIO(manim_stderr),
+                                    filename="Error.log",
+                                ),
+                            }
+
+                        return reply_args
```
[the remark this answers is not in the document]
Comment (Aathish04) [reply cogs/manimate.py]: And Done :)

File: cogs/mdoc.py [hunk 3]
```diff
@@ -26,23 +31,36 @@ async def mdoc(self, ctx, *args):
...
+                )
+        except errortype as e:
             if "AssertionError" in e.args[0]:
                 await ctx.reply(f"I could not find `{arg}` in our documentation, sorry.")
                 return
             await ctx.reply(f"Something went wrong: ```{e.args[0]}```")
             return
 
-        fqname = container_output.decode("utf-8").strip().splitlines()[2]
+        fqname = (proc.stdout.decode("utf-8").strip().splitlines()[0] if config.NO_DOCKER else container_output.decode("utf-8").strip().splitlines()[2])
```
Comment (behackl) [inline cogs/mdoc.py:63]:
```suggestion
        if config.NO_DOCKER:
            fqname = proc.stdout.decode("utf-8").strip().splitlines()[0]
        else:
            fqname = container_output.decode("utf-8").strip().splitlines()[2]
```

Again, being a bit more explicit does not hurt here.

**Review by behackl — APPROVED** [review]
Thank you; your changes look good to me. I have added some additional minor suggestions to improve readability; please take a look. Otherwise, I'm happy to merge this.

**Pull request merged**